src/vs/workbench/services/telemetry/browser/telemetryService.ts

```diff
@@ -50,6 +52,11 @@ export class TelemetryService extends Disposable implements ITelemetryService {
 				appenders.push(telemetryProvider);
 			}
 			appenders.push(new TelemetryLogAppender(loggerService, environmentService));
+			if (remoteAgentService.getConnection() !== null) {
+				const telemetryProvider: ITelemetryAppender = new GitpodInsightsAppender(productService);
+				appenders.push(telemetryProvider);
+				appenders.push(new TelemetryLogAppender(loggerService, environmentService));
+			}
```

Comment:
> Hey @jeanp413 , I'm thinking about why we need this if condition since, we are only browser here. Like @akosyakov said https://github.com/gitpod-io/openvscode-server/pull/376#discussion_r912714084

I'm not sure, did you tested that if there's a connection events are send using the backend and if connection is lost events are send using the webappender instead?

Let's test it this way first
```suggestion
			appenders.pop();
			const gitpodTelemetryProvider: ITelemetryAppender = remoteAgentService.getConnection() !== null ? { log: remoteAgentService.logTelemetry.bind(remoteAgentService), flush: remoteAgentService.flushTelemetry.bind(remoteAgentService) } : new GitpodInsightsAppender(productService);
			appenders.push(gitpodTelemetryProvider);
			appenders.push(new TelemetryLogAppender(loggerService, environmentService));

```

Reply:
Code like this will never create a Gitpod telemetry

Reply:
I think we can remove `if condition`

Reply:
what do you mean? that the events won't be send? then that means they are send to the backend so we need to move [all the events](https://github.com/gitpod-io/openvscode-server/blob/57d3840d9d89487462a2004551d58563022a7324/src/vs/gitpod/node/gitpodInsightsAppender.ts#L187-L350) to a file in a common folder, the new events too and import this new file in both the gitpodinsighappender in node and browser

Reply:
Could you sync and sort it out please? 🙏 In other regards it looks good to me.

Reply:
We only want to handle these logs in the browser itself.

Once we lose the conn, it means we can't connect to the server `remoteAgentService`, so the logs to the server will be lost. And this class will not be called twice, which means it will only be sent to the browser when the user refreshes the page, which is not expected.

In fact, what we need is to send to the server if we can connect to the server, and go to the new `GitpodInsighAppender` if we not.

Reply:
@jeanp413 I move `if` inside function

Reply:
I see so they are missing those events in upstream too? :sweat_smile: